turbopack/crates/turbopack-core/src/traced_asset.rs

```diff
@@ -32,9 +39,45 @@ impl OutputAsset for TracedAsset {
 
     #[turbo_tasks::function]
     async fn references(&self) -> Result<Vc<OutputAssets>> {
+        static SHARP_BINARY_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+            Regex::new("node_modules/@img/sharp-\\w+-\\w+/lib/sharp-(\\w+-\\w+).node$").unwrap()
+        });
+        let module_path = self.module.ident().path().await?;
+
+        // so sharp does require("@img/sharp-${arch}/sharp.node")
+        // @img/sharp-darwin-arm64/lib/sharp-darwin-arm64.node uses the dylib loading mechanism to
+        // load @img/sharp-libvips-darwin-arm64/libvips.dylib
+
+        let additional = if SHARP_BINARY_REGEX.is_match(&module_path.path) {
+            let arch = SHARP_BINARY_REGEX
+                .captures(&module_path.path)
```

Comment:
would `if let Some(captures) = SHARP_BINARY_REGEX.captures(...) {...` be better? only match the regex once?

Reply:
The idea was that the hot path is the regex not matching.
So to rather make that hot path slightly faster and reexecute once for that single matching module.

Reply:
makes sense, thanks.   I now recall reading the the regex crate was optimized for this exact case (no captures)